phpunit/functional/Glpi/Migration/GenericobjectPluginMigrationTest.php

```diff
@@ -0,0 +1,804 @@
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        parent::setUp();
+
+        // Load it inside the test DB transaction to not have to clean it manually
+        $queries = $DB->getQueriesFromFile(sprintf('%s/tests/fixtures/genericobject-migration/glpi-data.sql', GLPI_ROOT));
+        foreach ($queries as $query) {
+            $DB->doQuery($query);
+        }
+    }
+
+    public function testProcessMigration(): void
+    {
+        // Arrange
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        $migration = new GenericobjectPluginMigration($DB);
+        $result    = new PluginMigrationResult();
+        $this->setPrivateProperty($migration, 'result', $result);
+
+        // Act
+        $this->assertTrue($this->callPrivateMethod($migration, 'processMigration'));
+
+        // Assert
+
+        // Validate created asset definitions
+        $expected_definitions = [
+            [
+                'label'          => 'inactive',
+                'comment'        => 'Inactive main object.',
+                'icon'           => null,
+                'picture'        => null,
+                'is_active'      => false,
+                'capacities'     => [],
+                'profiles'       => [
+                    1 => 0,
+                    2 => 0,
+                    3 => 0,
+                    4 => 0,
+                    5 => 0,
+                    6 => 0,
+                    7 => 0,
+                    8 => 0,
+                ],
+                'translations'   => [],
+                'fields_display' => [
+                    ['key' => 'name', 'order' => 0, 'field_options' => []],
+                    ['key' => 'comment', 'order' => 1, 'field_options' => []],
+                ],
+                'date_creation'  => '2025-03-06 14:30:12',
+                // `date_mod` is automatically changed during the migration, we cannot prevent it
+                // 'date_mod'       => '2025-03-06 14:32:48',
+            ],
+            [
+                'label'          => 'tablet',
+                'comment'        => 'Main object with only than the mandatory fields.',
+                'icon'           => null,
+                'picture'        => null,
+                'is_active'      => true,
+                'capacities'     => [],
+                'profiles'       => [
+                    1 => 0,
+                    2 => 0,
+                    3 => 0,
+                    4 => 127,
+                    5 => 0,
+                    6 => 0,
+                    7 => 0,
+                    8 => 0,
+                ],
+                'translations'   => [],
+                'fields_display' => [
+                    ['key' => 'name', 'order' => 0, 'field_options' => []],
+                    ['key' => 'comment', 'order' => 1, 'field_options' => []],
+                ],
+                'date_creation'  => '2025-03-05 16:32:43',
+                // 'date_mod'       => '2025-03-06 09:58:55',
```

Comment:
Is this commented by mistake ?

Reply:
The problem here is that the migration must be made in multiple steps. The definition is first created, and later, the custome fields options are set by an update operation. There is no mechanism in GLPI that prevents the `date_mod` to be updated.
I wanted to validate that the imported `date_mod` was identical to the initial data, but it is not possible.

I have kept this comment to open the discussion on this very subject.